Sources/edge/Commands/Devices.swift

```diff
@@ -0,0 +1,152 @@
+import ArgumentParser
+import Foundation
+import Logging
+
+protocol DeviceDiscovery {
+    func findUSBDevices(logger: Logger) async -> [USBDevice]
+    func findEthernetInterfaces(logger: Logger) async -> [EthernetInterface]
+}
+
+struct DevicesCommand: AsyncParsableCommand {
+    static let configuration = CommandConfiguration(
+        commandName: "devices",
+        abstract: "List USB and Ethernet devices connected to the system"
+    )
+
+    enum DeviceType: String, ExpressibleByArgument {
+        case usb, ethernet, all
+    }
+
+    enum OutputFormat: String, ExpressibleByArgument {
+        case json, text
+    }
+
+    @Option(help: "Device types to list (usb, ethernet, or both)")
+    var type: DeviceType = .all
+
+    @Flag(name: [.customShort("j"), .long], help: "Output in JSON format")
+    var json: Bool = false
```

Comment:
Do we want to specify the format as an enum? Like `--format json`?

Reply:
I went that route but I didn't like it so changed to this. It's also more common to have --json